diff --git a/Makefile b/Makefile
index cc19a9d..3faa658 100644
--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,12 @@ ifeq ($(platform), sf2000)
     override CFLAGS += $(MIPS_FLAGS)
     override CXXFLAGS += $(MIPS_FLAGS) -fno-use-cxa-atexit -fno-exceptions -fno-rtti
     STATIC_LINKING = 1
+else ifeq ($(platform), sf3000)
+    TARGET = $(NAME)_libretro.so
+    override CFLAGS += -EL -mips32r2 -mfp32 -mhard-float -fPIC -G0 -O2 -Wno-incompatible-pointer-types -Wno-implicit-function-declaration
+    override CXXFLAGS += -EL -mips32r2 -mfp32 -mhard-float -fPIC -G0 -O2 -fpermissive -fno-exceptions -fno-rtti
+    override CFLAGS += -DFROGGY_SD_ROOT='"/mnt/sdcard"'
+    override CXXFLAGS += -DFROGGY_SD_ROOT='"/mnt/sdcard"'
 else
     TARGET = $(NAME)_libretro.so
     CC = gcc
diff --git a/pspkvm/platform_gb300/audio.c b/pspkvm/platform_gb300/audio.c
index 78ad32a..097a225 100644
--- a/pspkvm/platform_gb300/audio.c
+++ b/pspkvm/platform_gb300/audio.c
@@ -110,6 +110,7 @@ static fluid_synth_t     *g_fl_synth = NULL;
 static int                g_fl_probed = 0;
 
 static const char *g_fl_soundfonts[] = {
+    FROGGY_SD_ROOT "/cubegm/bios/j2me.sf2",
     "/mnt/sda1/bios/Roland_SC-55.sf2",
     "/mnt/sda1/system/Roland_SC-55.sf2",
     "/home/Sajnaps/gb300/bios/Roland_SC-55.sf2",
diff --git a/pspkvm/platform_gb300/platform.c b/pspkvm/platform_gb300/platform.c
index 5024432..b1a789c 100644
--- a/pspkvm/platform_gb300/platform.c
+++ b/pspkvm/platform_gb300/platform.c
@@ -37,6 +37,9 @@ extern void dly_tsk(int ms);
 
 #include <stdarg.h>
 __attribute__((weak)) void xlog(const char *fmt, ...) {
+    static int logging=-1;
+    if(logging<0)logging=access(FROGGY_SD_ROOT "/log.txt",F_OK)==0;
+    if(!logging)return;
     va_list ap;
     va_start(ap, fmt);
     vprintf(fmt, ap);
